vllm/v1/engine/core.py

```diff
@@ -131,6 +135,9 @@ def add_request(self, request: EngineCoreRequest):
                 request.mm_inputs, request.mm_hashes)
 
         req = Request.from_engine_core_request(request)
+        if req.use_guided_decoding:
+            # Start grammar compilation asynchronously
+            self.guided_decoding_manager.should_cache(req)
```

Comment:
We should consider renaming this. It returns a `bool`, but we don't use it. Maybe `warmup_cache()` or `populate_cache()` or something. What we're really saying is "check if you have a grammar cached already and if not, get that started!"

Reply:
resolved